pages/advanced/performance.es-ES.mdx

```diff
@@ -0,0 +1,122 @@
+# Rendimiento
+
+SWR proporciona una funcionalidad crítica en todo tipo de aplicaciónes web, por lo que el **rendimiento**
+es una prioridad absoluta.
+
+SWR’s tiene **caché** incorporado y la **[deduplicación](/advanced/performance#deduplication)** evitan las solicitudes de red innecesarias,
+pero el rendimiento del propio hook `useSWR` sigue siendo importante. En una aplicación compleja, puede haber cientos de llamadas `useSWR`
+en una solo página.
+
+SWR asegura que su aplicación tiene:
+
+- _no hay peticiones de red innecesarias_
+- _no hay renderizado innecesarios_
+- _no se importa código innecesario_
+
+sin ningún cambio de código por su parte.
+
+## Deduplicación
+
+Es muy común reutilizar los hooks SWR en tu aplicación. Por ejemplo, una aplicación que muestra el avatar del usuario
+actual 5 veces:
+
+```jsx
+
+function useUser() {
+   return useSWR('/api/user', fetcher)
+}
+
+function Avatar () {
+   const { data, error } = useUser()
+   if (error) return <Error />
+   if (!data) return <Spinner />
+
+   return <img src={data.avatar_url} />
+}
+
+function App() {
+  return <>
+    <Avatar />
+    <Avatar />
+    <Avatar />
+    <Avatar />
+    <Avatar />
+  </>
+}
+
+```
+
+Cada componente `<Avatar/>` tiene un hook `useSWR` en su interior. Dado que tienen el mismo key SWR y 
+que se renderizan casi al mismo tiempo, **sólo se hará 1 solicitud de red**.
+
+Puedes reutilizar tus hooks de datos (como `useUser` en el ejemplo anterior) en todas partes, sin preocuparte por el rendimiento
+o las peticiones duplicadas.
+
+También existe la [opción `dedupingInterval`](/docs/options) para anular el intervalo de deduplicación por defecto.
+
+## Comparación profunda
+
+SWR por defecto tiene  **deep compares** al cambio de datos. Si el valor de `data` no ha cambiado, no se
+activará una nueva renderización.
+
+También puede personalizar la función de comparación mediante la [opción `compare`](/docs/options) si quieres cambiar el comportamiento.
+Por ejemplo, algunas respuestas de la API devuelven una marca de tiempo del servidor que tal vez quiera excluir de la difusión de datos.
+
+## Coleción de dependencias
+
+`useSWR` devuelve 3 valores de **estado**: `data`, `error` y `isValidating` cada uno de ellos puede actualizarse de forma independientemente.
+Por ejemplo, si imprimimos esos valores dentro de un ciclo de vida completo de obtención de datos, será algo como esto:
+
+```jsx
+
+function App () {
+  const { data, error, isValidating } = useSWR('/api', fetcher)
+  console.log(data, error, isValidating)
+  return null
+}
+
+```
+
+En el peor de los casos (la primera solicitud falló, entonces el reitento fue exitoso), verá 4 líneas de registros:
```

Comment:
typo: reintento

Reply:
use 'entonces' as in 'then' not 'therefore'. "Si la primera solicitud falló, entonces el reintento fue exitoso"

Reply:
verá is formal you. Let's use impersonal: "Se verán 4 líneas de registros"